src/main/scala/chisel3/experimental/inlinetest/InlineTest.scala

```diff
@@ -178,40 +279,44 @@ trait HasTests { module: RawModule =>
   private lazy val moduleDefinition = module.toDefinition.asInstanceOf[Definition[M]]
 
   /** Generators for inline tests by name. LinkedHashMap preserves test insertion order. */
-  private val registeredTests = new mutable.LinkedHashMap[String, RegisteredTest[M, _]]
+  private val registeredTests = new mutable.LinkedHashMap[String, RegisteredTest[M]]
 
   /** Get the currently registered tests for this module and whether they are queued for elaboration. */
-  private def getRegisteredTests: Seq[RegisteredTest[M, _]] =
+  private def getRegisteredTests: Seq[RegisteredTest[M]] =
     registeredTests.values.toSeq
 
   /** Get all enabled tests for this module. */
-  def getTests: Seq[TestParameters[M, _]] =
+  def getTests: Seq[TestParameters[M]] =
     getRegisteredTests.filter(_.shouldElaborateToCircuit).map(_.params)
 
   /** Map from test name to elaborated test. */
-  private val elaboratedTests = new mutable.LinkedHashMap[String, ElaboratedTest[M, _]]
+  private val elaboratedTests = new mutable.LinkedHashMap[String, ElaboratedTest[M]]
 
   /** Get the all tests elaborated to the ciruit. */
-  private[chisel3] def getElaboratedTests: Seq[ElaboratedTest[M, _]] =
+  private[chisel3] def getElaboratedTests: Seq[ElaboratedTest[M]] =
     elaboratedTests.values.toSeq
 
   /** Elaborate a test to the circuit as a public definition. */
-  private def elaborateTestToCircuit[R](test: RegisteredTest[M, R]): Unit =
+  private def elaborateTestToCircuit[R](test: RegisteredTest[M]): Unit =
     Definition {
       val elaboratedTest = test.elaborate()
       elaboratedTests += test.params.testName -> elaboratedTest
       elaboratedTest.testHarness
     }
 
+  /** Get all tests that were elaborated. */
+  private[chisel3] def getElaboratedTestModules: Seq[ElaboratedTest[M]] =
+    elaboratedTests.values.toSeq
+
   /** Generate a public module that instantiates this module. The default
     *  testharness has clock and synchronous reset IOs and contains the test
     *  body.
     *
     *  @param testBody the circuit to elaborate inside the testharness
     */
-  protected final def test[R](
+  protected final def test(
     testName: String
-  )(testBody: Instance[M] => R)(implicit testHarnessGenerator: TestHarnessGenerator[M, R]): Unit = {
+  )(testBody: Instance[M] => TestConfiguration)(implicit testHarnessGenerator: TestHarnessGenerator[M]): Unit = {
```

Comment:
At first glance, it seems like the test configuration is a parameter of the `test` method itself and not the return of the lambda. I.e., it seems more natural for this to be:

```
protected def test(testName: String, config: TestConfiguration)(body: Instance[R] => Unit)
```

However, the PR is using this to force the communication of how the success and finish are hooked up.

I don't see anything wrong here, just this is unexpected.